File: service-backbase-java/src/main/java/org/example/backend/config/filter/RequestIdFilter.java
package org.example.backend.config.filter;

import java.util.UUID;
import pl.net.testit.serum.api.filter.RequestHeaderFilter;

public class RequestIdFilter extends RequestHeaderFilter {

  public static final String HEADER_NAME = "Request-Id";

  public RequestIdFilter() {
    super(HEADER_NAME);
    this.setValue(() -> "|" + UUID.randomUUID() + ".");
  }
}

File: service-backbase-kotlin/src/main/kotlin/org/example/backend/config/filter/RequestIdFilter.kt
package org.example.backend.config.filter

import pl.net.testit.serum.api.filter.RequestHeaderFilter
import java.util.*

class RequestIdFilter : RequestHeaderFilter("Request-Id") {
  init {
    this.setValue { "|" + UUID.randomUUID() + "." }
  }
}